docs/specs/ops.yml

```diff
@@ -282,3 +282,82 @@ outputs:
   .errors.log: |
     {"message_severity":"error","code":"violate-schema","message":"Error converting value \"a/b/.openpublishing.redirection.test.json\" to type 'System.Collections.Generic.HashSet`1[Microsoft.Docs.Build.PathString]'.","file":".openpublishing.publish.config.json","line":3,"column":66}
     {"message_severity":"error","code":"violate-schema","message":"Error converting value \"a/b/.openpublishing.redirection.test.json\" to type 'System.Collections.Generic.HashSet`1[Microsoft.Docs.Build.PathString]'.","file":".openpublishing.publish.config.json","line":3,"column":66}
+
+---
+# provide source url in .publish.json for content, toc and resource pages but not redirection pages.
+repos:
+  https://github.com/test/repo:
+  - files:
+      .openpublishing.publish.config.json: |
+        {
+          "docsets_to_publish": [{"build_source_folder":"a"}],
+        }
+      .openpublishing.redirection.json: |
+        {
+          "redirections": [
+            {"source_path": "a/missing.md", "redirect_url": "/a/a"},
+          ]
+        }
+      a/docfx.yml:
+      a/a.md:
+      a/toc.yml:
+      a/image.png:
+outputs:
+   a/a.json:
+   a/image.png:
+   a/toc.json:
+   a/.publish.json: |
+    {
+      "files": [
+        {
+            "source_path": "missing.md",
+            "source_url": undefined
+        },
+        {
+            "source_path": "a.md",
+            "source_url": "https://github.com/test/repo/blob/main/a/a.md"
+        },
+        {
+            "source_path": "image.png",
+            "source_url": "https://github.com/test/repo/blob/main/a/image.png"
+        },
+        {
+            "source_path": "toc.yml",
+            "source_url": "https://github.com/test/repo/blob/main/a/toc.yml"
+        }
+      ]
+    }
+---
+# source url also works for CRR
+repos:
+  https://github.com/ops1/repo:
+  - files:
+      .openpublishing.publish.config.json: |
+        {
+          "docsets_to_publish": [
+            { "build_source_folder": "articles" },
+          ],
+          "dependent_repositories": [
```

Comment:
Search yml specs for `dependencies:` to see how to configure dependency repos using `docfx.yml`. Use openpublishing.xyz file only for docs/ops specific features.